tests/parametric/test_trace_sampling.py

```diff
@@ -348,6 +394,175 @@ def test_trace_dropped_by_trace_sampling_rule_tags(self, test_agent, test_librar
+class Test_Trace_Sampling_Tags_Feb2024_Revision:
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            tag_sampling_env("foo"),
+            tag_sampling_env("fo*"),
+            tag_sampling_env("f??"),
+            tag_sampling_env("?o*"),
+            tag_sampling_env("???"),
+            tag_sampling_env("*"),
+            tag_sampling_env("**"),
+        ],
+    )
+    def test_globs_same_casing(self, test_agent, test_library):
+        """Test tag matching with string of matching case"""
+        with test_library:
+            with test_library.start_span(name="matching-span", service="test") as span:
+                span.set_meta("tag", "foo")
+
+        matching_span = find_span_in_traces(
+            test_agent.wait_for_num_traces(1), Span(name="matching-span", service="test")
+        )
+
+        assert matching_span["metrics"].get(SAMPLING_PRIORITY_KEY) == 2
+        assert matching_span["metrics"].get(SAMPLING_RULE_PRIORITY_RATE) == 1.0
+
+    @pytest.mark.parametrize(
+        "library_env",
+        [tag_sampling_env("Foo"), tag_sampling_env("Fo*"), tag_sampling_env("F??"), tag_sampling_env("?O*")],
+    )
+    def test_globs_different_casing(self, test_agent, test_library):
+        """Test tag matching with string of matching case"""
+        with test_library:
+            with test_library.start_span(name="matching-span", service="test") as span:
+                span.set_meta("tag", "foo")
+
+        matching_span = find_span_in_traces(
+            test_agent.wait_for_num_traces(1), Span(name="matching-span", service="test")
+        )
+
+        assert matching_span["metrics"].get(SAMPLING_PRIORITY_KEY) == 2
+        assert matching_span["metrics"].get(SAMPLING_RULE_PRIORITY_RATE) == 1.0
+
+    @pytest.mark.parametrize("library_env", [tag_sampling_env("[abc]")])
+    def test_no_set_support(self, test_agent, test_library):
+        """Test verifying that common glob set extension is NOT supported"""
+        with test_library:
+            with test_library.start_span(name="matching-span", service="test") as span:
+                span.set_meta("tag", "[abc]")
+
+        matching_span = find_span_in_traces(
+            test_agent.wait_for_num_traces(1), Span(name="matching-span", service="test")
+        )
+
+        assert matching_span["metrics"].get(SAMPLING_PRIORITY_KEY) == 2
+        assert matching_span["metrics"].get(SAMPLING_RULE_PRIORITY_RATE) == 1.0
+
+    @pytest.mark.parametrize("library_env", [tag_sampling_env("[a-c]")])
+    def test_no_range_support(self, test_agent, test_library):
+        """Test verifying that common glob range extension is NOT supported"""
+        with test_library:
+            with test_library.start_span(name="matching-span", service="test") as span:
+                span.set_meta("tag", "[a-c]")
+
+        matching_span = find_span_in_traces(
+            test_agent.wait_for_num_traces(1), Span(name="matching-span", service="test")
+        )
```

Comment:
Could we extract these lines to a common method in the class? They are constantly repeated in the different methods, varying only: `span.set_meta("tag", "XYZ")`

Reply:
I'd be fine with.  I debated whether or not that would obscure the test too much.